android/src/com/mapswithme/maps/widget/placepage/PlacePageView.java

```diff
@@ -801,6 +803,20 @@ private void refreshViewsInternal(@NonNull MapObject mapObject)
     }
   }
 
+  private Spanned getShortDescription(@NonNull MapObject mapObject)
+  {
+    String htmlDescription = mapObject.getDescription();
+    final int paragraphStart = htmlDescription.indexOf("<p>");
+    final int paragraphEnd = htmlDescription.indexOf("</p>");
+    if (paragraphStart == 0)
+      htmlDescription = htmlDescription.substring(3, paragraphEnd);
```

Comment:
1. Why is it needed?
2. `</p>` is not removed

Reply:
1. It is the way to get first paragraph. I was thinking about two solutions:
  * The first one presented.
  * To run `Html.fromHtml()` first and then substring first paragraph, but `fromHtml()` returns `Spanned` that don't have any find methods. This solution will be much complex and, I believe, much slower.
2. `</p>` is removed. `indexOf("</p>")` returns the index of the beginning of this tag(in this case, index of '<' symbol), and range in `substring` is passed in the form `[from, to)`

Reply:
@biodranik What do you think?

Reply:
1. Is it guaranteed that `<p>` is always present?
2. Is it guaranteed that it's always `<p>`, and never `<P>`?
3. Is it guaranteed that the text between `<p>` and `</p>` is always the first paragraph?

Reply:
Yes, yes and yes. Description is fetched from Wikipedia by `tools/python/descriptions` script and it uses Wikipedia API, here is a [link](https://github.com/martin-majlis/Wikipedia-API#how-to-get-full-text) with usage example and result it returns.

1. As in the example, summary section always contains `<p>`.
2. As it is an HTML text processed by tons of parsers and converters before it appears on our side, I think, we can say "yes, it is guaranteed".
3.  See 1st point.